src/DataAccess/TicketNumberReader.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\OtrsExtractAddress\DataAccess;
+
+/**
+ * @license GNU GPL v2+
+ * @author Gabriel Birke < gabriel.birke@wikimedia.de >
+ */
+class TicketNumberReader extends \IteratorIterator {
+
+	const TICKET_NUMBER_COLUMN = 0;
+
+	public function __construct( \SplFileObject $inputFile ) {
+		$inputFile->setCsvControl( ';', '"' );
+		$inputFile->setFlags( \SplFileObject::READ_CSV | \SplFileObject::READ_AHEAD | \SplFileObject::SKIP_EMPTY
+			| \SplFileObject::DROP_NEW_LINE );
+		parent::__construct( $inputFile );
```

Comment:
This modification of something passed by-ref in the constructor is kinda unexpected. If another service also gets the same object and modifies it differently things go badly.

Since this class binds tightly to the SplFileObject anyway, perhaps that thing can be constructed in the constructor here? Looking at the users that is problematic for the nice in-memory unit test... 

This problem is coming from one object having both the contents and the way how that contents is read. Perhaps the API does not force that though? (I can't ssh on this network so hard to check stuff)